eclass/toolchain.eclass

```diff
@@ -1635,10 +1635,20 @@ gcc_do_filter_flags() {
 
 	# These are set here so we have something sane at configure time
 	if is_crosscompile ; then
+		local orig_CFLAGS="${CFLAGS}"
+
 		# Set this to something sane for both native and target
 		CFLAGS="-O2 -pipe"
-		FFLAGS=${CFLAGS}
-		FCFLAGS=${CFLAGS}
+		FFLAGS="${CFLAGS}"
+		FCFLAGS="${CFLAGS}"
+
+		# If the original CFLAGS set by the user have debugging
+		# symbol settings in them, let's try to (weakly) preserve
+		# that. It'll make debugging cross-compilers easier.
+		# bug #792321.
+		if [[ "${orig_CFLAGS}" =~ -g?(gdb)?([1-9]) ]] ; then
```

Comment:
```suggestion
		if [[ "${orig_CFLAGS}" =~ -g?(gdb)?(lldb)?([1-9])?$ ]] ; then
```

How about this? Also matches to `-glldb` and plain `-g`: https://regex101.com/r/nHgVG6/1

Reply:
I feel like could use a new flag-o-matic function or something for debug flags, gets copy/pasted in all sort of different ways for testing if debug is used and of course misses things like that, not to say function is necessary to handle this one (plus this need to keep the flag rather than just a test).

Also did say it's untested yet :) This uses extglob (not enabled) and needs == (which gives no bash rematch).

fwiw think this "would" do what's intended here, and won't match -gfoo.
```
if [[ ${orig_CFLAGS} =~ (\ |^)(-g(gdb|lldb)?[0-9]?)(\ |$) ]]; then
    CFLAGS+=" ${BASH_REMATCH[2]}"
fi
```
..but a non-loop approach it won't pickup multiple flags, e.g. extract the two -g from `-g -O2 -ggdb3`, nor understand that debug is disabled if there's `-g0` at the end.

flag-o-matic tend to do word splitting instead, aka:
```
eshopts_push -s extglob
for flag in ${orig_CFLAGS}; do
        [[ ${flag} == -g?(gdb)?(lldb)?([0-9]) ]] && CFLAGS+=" ${flag}"
done
eshopts_pop
```
(includes 0 for the above case)

Just slapped these together so I may overlooking something

Reply:
Need not to forget inherit estack for the latter too, see toolchain.eclass doesn't have it.